src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -108,9 +111,11 @@ private function createType(DocType $docType): ?Type
 
         if ('mixed[]' === $docTypeString) {
             $docTypeString = 'array';
+        } elseif ('array' !== $docTypeString && $docType instanceof Array_ && null === new \ReflectionProperty(Array_::class, 'keyType')->getValue($docType)) {
+            $docTypeString = sprintf('%s[]', $docType->getValueType());
         }
 
-        if ($docType instanceof Collection) {
+        if ($docType instanceof Collection || $docType instanceof Generic) {
```

Comment:
`Collection` got replaced with `Generic` in phpDocumentor/TypeResolver#238

Reply:
Collection and generic are not compatible, generic  can be any generic even when it's not a collection like. The previous versions had this same possible miss, but it was less common as the collection had the requirement of 2 values. Like `collection<key, value>`. If this didn't match it would return an invalid tag definition. 

The problem is that this dated from before generics existed. Today a Generic can have any number of templated types. 

To be closer to the actual collection you might want to check for the number of types in the generic. Which doesn't necessarily mean it's a collection, but that's probably a problem that existed in earlier versions.